File: app/src/main/java/net/programmierecke/radiodroid2/station/live/metadata/TrackMetadataCallback.java
package net.programmierecke.radiodroid2.station.live.metadata;


import androidx.annotation.NonNull;

public interface TrackMetadataCallback {
    enum FailureType {
        RECOVERABLE,
        UNRECOVERABLE,
    }

    void onFailure(@NonNull FailureType failureType);
    void onSuccess(@NonNull TrackMetadata trackMetadata);
}

File: app/src/main/java/net/programmierecke/radiodroid2/station/live/metadata/TrackMetadataCallback.kt
package net.programmierecke.radiodroid2.station.live.metadata

interface TrackMetadataCallback {
    enum class FailureType {
        RECOVERABLE, UNRECOVERABLE
    }

    fun onFailure(failureType: FailureType)
    fun onSuccess(trackMetadata: TrackMetadata)
}
